server/transcribe-server.py

```diff
@@ -1,42 +1,115 @@
+app.config["MAX_CONTENT_LENGTH"] = 3 * 1024 * 1024  # 1 Megabyte
 
-from threading import Semaphore
 
 MAX_CONCURRENT_TASKS = 2
 task_semaphore = Semaphore(MAX_CONCURRENT_TASKS)
 
 MAX_IN_FLIGHT_TASKS = 5000
 in_flight_semaphore = Semaphore(MAX_IN_FLIGHT_TASKS)
 
-whisper_models = [] 
+whisper_models = []
+
+
+@app.route("/v1/audio/transcriptions", methods=["POST"])
+@app.route("/audio/transcriptions", methods=["POST"])
+def transcribe_audio():
+    try:
+        request_data = CreateTranscriptionRequest(
+            **{
+                "file": request.files["file"].stream.read(),
+                "model": request.form.get("model", "whisper-1"),
+                "response_format": request.form.get("response_format", "json"),
+                "language": request.form.get("language"),
+                "prompt": request.form.get("prompt"),
+            }
+        )
+
+        if request.form.get("temperature") is not None:
+            request_data.temperature = float(request.form.get("temperature"))
+        if request.form.get("timestamp_granularities[]") is not None:
+            request_data.timestamp_granularities = request.form.getlist(
+                "timestamp_granularities[]"
+            )
+    except Exception as e:
+        return str(e), 400
+
+    if not in_flight_semaphore.acquire(blocking=False):
+        return (
+            jsonify({"error": "Too many tasks in flight"}),
+            429,
+        )  # HTTP 429 Too Many Requests
 
-@app.route('/execute', methods=['POST'])
+    try:
+        # Process the task
+        with task_semaphore:
+            model = None
+            try:
+                # Model parameter is ignored - we use a predefined model at this point
+                model = whisper_models.pop()
+
+                segments, transcription_info = model.transcribe(
+                    io.BytesIO(request_data.file),
+                    task="transcribe",
+                    language=request_data.language,
+                    initial_prompt=request_data.prompt,
+                    word_timestamps="word" in request_data.timestamp_granularities,
+                    temperature=request_data.temperature,
+                )
+
+                # Format the response
+                response = segments_to_response(
+                    segments, transcription_info, request_data.response_format
+                )
+                if request_data.response_format == ResponseFormat.TEXT:
+                    return response, 200, {"Content-Type": "text/plain"}
+
+                response_json = response.model_dump_json()
+                return response_json, 200, {"Content-Type": "application/json"}
+
+            except Exception as e:
+                return jsonify({"error": str(e)}), 500
+
+            finally:
+                if model is not None:
+                    whisper_models.append(model)
+
+    finally:
+        in_flight_semaphore.release()
+
+
+@app.route("/execute", methods=["POST"])
```

Comment:
I think this path should be removed.

No one should be using it if we use the new format, and using two different endpoint types means more code to maintain and more questions to answer.

Your new code is the way to go.